internal/services/cdn/cdn_frontdoor_firewall_policy_data_source.go

```diff
@@ -66,42 +66,59 @@ func dataSourceCdnFrontDoorFirewallPolicy() *pluginsdk.Resource {
 }
 
 func dataSourceCdnFrontDoorFirewallPolicyRead(d *pluginsdk.ResourceData, meta interface{}) error {
-	client := meta.(*clients.Client).Cdn.FrontDoorLegacyFirewallPoliciesClient
+	client := meta.(*clients.Client).Cdn.FrontDoorFirewallPoliciesClient
 	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
 	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	id := parse.NewFrontDoorFirewallPolicyID(subscriptionId, d.Get("resource_group_name").(string), d.Get("name").(string))
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
 
-	resp, err := client.Get(ctx, id.ResourceGroup, id.FrontDoorWebApplicationFirewallPolicyName)
+	id := waf.NewFrontDoorWebApplicationFirewallPolicyID(subscriptionId, resourceGroup, name)
+
+	result, err := client.PoliciesGet(ctx, id)
 	if err != nil {
-		if utils.ResponseWasNotFound(resp.Response) {
+		if !response.WasNotFound(result.HttpResponse) {
 			return fmt.Errorf("%s was not found", id)
 		}
 
 		return fmt.Errorf("retrieving %s: %+v", id, err)
 	}
 
+	model := result.Model
+
+	if model == nil {
+		return fmt.Errorf("retrieving %s: 'model' was nil", id)
+	}
```

Comment:
The pattern we've been adopting for `Read` when migrating to `go-azure-sdk` is to do all the `d.Set()` etc inside the nil check for he model, e.g.
```
	if model := result.Model; model != nil {
		if props := model.Properties; props != nil {
			// TODO all the `d.Set("props_thing", props.Thing)` stuff
		}
		// TODO - all the `d.Set("top_level_prop", model.Prop)`
	}
```
Comprehensive example from AKS [here](https://github.com/hashicorp/terraform-provider-azurerm/blob/338dca6e89a579dde84c13f3498418e1c85d44ac/internal/services/containers/kubernetes_cluster_resource.go#L2566)

Reply:
Fixed.